### DISCUSSION TITLE
Google AMP Stories Support

### DISCUSSION THREAD

**Discussion by oguzbilgic** [vercel/next.js#20079] [Ideas]
Google recently introduced a product called [Google stories](http://stories.google/)

As far as I can understand, this implements AMP with special HTML tags. Eventually, this ends up being published as an HTML page with a bookend.json file attached. 

Not sure what would be the best way to implement this using next.js

**Comment by brunomarks7** [comment]
I have the same question. [This tool](https://makestories.io/) allows you to create, export, and upload by FTP. 

My CMS used in NextJS is WordPress, and it has an official plugin to create web-stories. However, WordPress uses an IP address and I don't know how SEO works with that.

**Comment by slavakurilyak** [comment]
@Google provides [SEO](https://developers.google.com/search/docs/advanced/appearance/web-stories-creation-best-practices#seo) and [technical](https://developers.google.com/search/docs/advanced/appearance/web-stories-creation-best-practices#technical) best practices for implementing web stories.

Google started testing `amp-script` (custom JS in a web worker) using popular JS frameworks:

> The team has tested amp-script with frameworks like Vue, Angular, Aurelia, and lit-html, but less extensively ([amp.dev](https://amp.dev/documentation/guides-and-tutorials/develop/custom-javascript/?format=websites#:~:text=the%20team%20has%20tested%20amp-script%20with%20frameworks%20like%20vue%2C%20angular%2C%20aurelia%2C%20and%20lit-html%2C%20but%20less%20extensively))

Google also recognizes JS bundle size limitations:

> That said, React bundles often exceed amp-script's 150K limit. So it's recommended that you use Preact, a lightweight alternative to React ([amp.dev](https://amp.dev/documentation/guides-and-tutorials/develop/custom-javascript/?format=websites#:~:text=that%20said%2C%20react%20bundles%20often%20exceed%20amp-script's%20150k%20limit.%20so%20it's%20recommended%20that%20you%20use%20preact%2C%20a%20lightweight%20alternative%20to%20react))

**Comment by raphaeljoer** [comment]
Check this guys! 
https://nextjs.org/docs/advanced-features/amp-support/adding-amp-components

**Reply by timofeysie** [reply]
> Check this guys! https://nextjs.org/docs/advanced-features/amp-support/adding-amp-components

@raphaeljoer , that's for  AMP components.  This question is about Google AMP Stories, which have a very specific format.  As @oguzbilgic pointed out, they are deployed with a bookend.json file attached.  A Next.js example or tutorial on how to author an AMP Story would work, or possibly an addition to the cli to generate a boilerplate version.

However, as @slavakurilyak pointed out, bundle size might be a limitation, and so Preact is recommended.  Does this mean that a regular Next.js app with AMP pages would not work for SEO which is the intended purpose of AMPs?

Since there is a [Preact demo](https://github.com/vercel/next.js/tree/canary/examples/using-preact) already, a version of this that is setup for AMPs and another version for AMP stories would be what is needed.

**Comment by devmoatassem** [comment]
@raphaeljoer can you please share any refrence for app router ?
